hidden-posts.php

```diff
@@ -32,6 +32,7 @@ function __construct() {
         add_action( 'post_submitbox_misc_actions', array( $this, 'hidden_checkbox' ) );
         add_action( 'save_post', array( $this, 'save_meta' ) );
         add_action( 'pre_get_posts', array( $this, 'pre_get_posts' ) );
+        add_action( 'upgrader_process_complete', array( $this, 'migrate_posts' ));
```

Comment:
> I do not think this hook works like what you want. It only works when you try to update through WP CLI or WP Admin.

> Otherwise, like me or our VIP customers via git or other ways to update plugins, this hook will not be fired.

I wasn't aware of that. Thanks for sharing this.

> Also, it's not ideal to fire this hook for every update. I think it should run only one time. A pattern for this is what I have seen here recently https://github.com/Automattic/vip-support/blob/1c47e4c9ae71c966c6c0c275e6f010e1f6fcba4d/class-vip-support-role.php#L158-L165

> You set an option and check its value to decide whether or not you run the immigration.

Yeah, that's a problem I also had in mind. Thanks for pointing out the pattern at `vip-support`. That's indeed a better approach than what I implemented.